service/s3/internal/customizations/update_endpoint.go

```diff
@@ -20,19 +26,48 @@ type UpdateEndpointOptions struct {
 
 	// use path style
 	UsePathStyle bool
+
+	// use transfer acceleration
+	UseAccelerate bool
+
+	// functional pointer to indicate support for accelerate.
+	// The function is intended to take an input value, and
+	// return if the operation supports accelerate.
+	SupportsAccelerate func(interface{}) bool
+
+	// use dualstack
+	UseDualstack bool
 }
 
 // UpdateEndpoint adds the middleware to the middleware stack based on the UpdateEndpointOptions.
 func UpdateEndpoint(stack *middleware.Stack, options UpdateEndpointOptions) {
```

Comment:
The `service/s3/internal/customization/doc.go` probably should be updated to describe this new feature(s) (s3control's too)